internal/plugin/service/service/crud.go

```diff
@@ -0,0 +1,325 @@
+		return
+	}
+
+	o.ServiceType = types.StringValue(c.serviceType)
+	flattenUserConfig(ctx, diags, o, s)
+	if diags.HasError() {
+		return
+	}
+
+	integrations := schemautil.FlattenSetNested(ctx, diags, flattenServiceIntegration, s.Integrations, serviceIntegrationAttrs)
+	if diags.HasError() {
+		return
+	}
+
+	tagsRsp, err := c.client.ServiceTags.Get(ctx, project, serviceName)
+	if err != nil {
+		diags.AddError("Unable to get service tags", err.Error())
+		return
+	}
+
+	tags := flattenTags(ctx, diags, tagsRsp.Tags)
+	if diags.HasError() {
+		return
+	}
+
+	staticIPsRsp, err := c.client.StaticIPs.List(ctx, project)
+	if err != nil {
+		diags.AddError(fmt.Sprintf("Unable to list static ips for project '%s'", project), err.Error())
+		return
+	}
+
+	staticIPs, d := types.SetValueFrom(ctx, types.StringType, staticIPsRsp.StaticIPs)
+	diags.Append(d...)
+	if diags.HasError() {
+		return
+	}
+
+	o.ID = types.StringValue(filepath.Join(project, serviceName))
+	o.CloudName = types.StringValue(s.CloudName)
+	o.MaintenanceWindowDow = types.StringValue(s.MaintenanceWindow.DayOfWeek)
+	o.MaintenanceWindowTime = types.StringValue(s.MaintenanceWindow.TimeOfDay)
+	o.Plan = types.StringValue(s.Plan)
+	o.ProjectVPCId = types.StringPointerValue(s.ProjectVPCID)
+	o.Project = types.StringValue(project)
+	o.ServiceHost = types.StringValue(s.URIParams["host"])
+	if p := s.URIParams["password"]; p != "" {
+		o.ServicePassword = types.StringValue(p)
+	}
+	if u := s.URIParams["user"]; u != "" {
+		o.ServiceUsername = types.StringValue(u)
+	}
+	port, err := strconv.ParseInt(s.URIParams["port"], 10, 32)
+	if err != nil {
+		diags.AddError("Can't parse port", err.Error())
+		return
+	}
+	o.ServicePort = types.Int64Value(port)
+	o.ServiceName = types.StringValue(serviceName)
+	o.ServiceURI = types.StringValue(s.URI)
+	o.ServiceIntegrations = integrations
+	o.State = types.StringValue(s.State)
+	o.StaticIPs = staticIPs
+	o.Tag = tags
+	o.TerminationProtection = types.BoolValue(s.TerminationProtection)
+
+	disk := c.diskSpace(ctx, diags, o)
+	if diags.HasError() {
+		return
+	}
+	o.DiskSpaceUsed = disk.String(s.DiskSpaceMB) // Actual service disk size
+	o.AdditionalDiskSpace = disk.String(s.DiskSpaceMB - disk.AdditionalDiskSpace())
+	o.DiskSpaceCap = disk.String(disk.plan.DiskSpaceCapMB)
+	o.DiskSpaceDefault = disk.String(disk.plan.DiskSpaceMB)
+	o.DiskSpaceStep = disk.String(disk.plan.DiskSpaceStepMB)
+}
+
+func (c *crud) diskSpace(ctx context.Context, diags diag.Diagnostics, o *Resource) *diskSpace {
+	plan, err := c.client.ServiceTypes.GetPlan(ctx, o.Project.ValueString(), c.serviceType, o.Plan.ValueString())
+	if err != nil {
+		diags.AddError("Error getting service default plan parameters", err.Error())
+		return nil
+	}
+	return &diskSpace{plan: plan, resource: o}
+}
+
+type diskSpace struct {
+	plan     *aiven.GetServicePlanResponse
+	resource *Resource
+}
+
+func (d *diskSpace) String(size int) types.String {
```

Comment:
it could be clearer if renamed to something more descriptive, like `BytesToString`